src/parse.cpp

```diff
@@ -264,10 +267,17 @@ expression* parser::parse_primary_expression(precedence prec) {
   // (x) => {}    // Arrow function.
   // (x + y * z)  // Parenthesized expression.
   case token_type::left_paren: {
+    this->depth_++;
```

Comment:
Blocking: Where is the matching `depth--` for inputs like `(x)`? It seems like you only handle `()`.

If I put `(x);` in a file 301 times, your patch causes quick-lint-js to fail. This is wrong.

I think we should use a scope guard to prevent mistakes like this. See `parser::function_guard` for an example.